api_tests/nodes/views/test_node_detail.py

```diff
@@ -1865,6 +1865,27 @@ def test_component_return_parent_license_if_no_license(
             API_BASE, node_license._id)
         assert expected_license_url in actual_license_url
 
+    def test_node_license_on_create(
```

Comment:
We need several more tests to properly capture the many invalid ways in which a payload could be formatted, and to make sure your serializer changes are properly handling all errors (your current work will return 500's for many scenarios).

For example, in addition to only the `node_license` portion of the request being specified, as you have here, I'd add a test for where only the `license` portion of the request is specified, and when both portions are specified.  What about if an invalid license is specified, or only `copyright_holders` is in the payload?

`api_tests/nodes/views/test_node_detail.py.TestNodeUpdateLicense` looks pretty thorough in how it's testing different license use cases.  You could follow a very similar format for NodeCreatewithLicense.